Sources/MapboxNavigation/NavigationTilesVersioningDelegate.swift

```diff
@@ -0,0 +1,40 @@
+
+import Foundation
+import MapboxCoreNavigation
+
+/**
+ Protocol for handling Navigation Tiles version switching.
+ 
+ Such events may occur during navigation with poor or absent Internet connectivity. Tiles versioning relies on pre-downloaded `OfflineRegion`s to contain suitable tiles.
+ */
+public protocol NavigationTilesVersioningDelegate: AnyObject, UnimplementedLogging {
+    /**
+     Called when navigation wants to switch Navigation Tiles version
+     
+     This happens when latest Navigation tiles are not available for current location (due to connectivity issues for example), but there is pre-downloaded offline data available which is older version.
+     
+     - parameter versions: An array of version names available for switching to. These are not designed to be displayed to User directly.
+     - parameter completion: Completion callback which takes one of the version names provided by `versions`. Sender will reconfigure the navigation engine immediately. You may ignore calling the callback at all if you choose to ignore the version switch. Keep in mind that such decision may result in poor user experience.
+     */
+    func tilesVersioning(wantsSwitchTilesVersion versions: [String], completion: @escaping (String?) -> ())
+    
+    /**
+     Called when navigation is able to swtich back to the latest tiles version.
+     
+     Usually happens after `tilesVersioning(wantsSwitchTilesVersion:, completion:)`. The process is the same hence there is no version choice.
+     
+     - parameter restoreCallback: A callback to perform or not the switching. You may ignore calling it if you choose to ignore the event.
+     */
+    func tilesVersioning(wantsRestoreTilesVersion restoreCallback: @escaping (Bool) -> ())
+}
+
+
+public extension NavigationTilesVersioningDelegate {
+    func tilesVersioning(wantsSwitchTilesVersion versions: [String], completion: @escaping (String?) -> ()) {
+        logUnimplemented(protocolType: NavigationTilesVersioningDelegate.self,  level: .info)
```

Comment:
Doesn't the default implementation do anything?

Reply:
I don't think we should do something in default implementation other than ignore the event. Forcing a version switch by default may result in unexpected UI hiccups for user